module/os/linux/zfs/zpl_xattr.c

```diff
@@ -706,19 +706,29 @@ static int
 __zpl_xattr_user_get(struct inode *ip, const char *name,
     void *value, size_t size)
 {
-	char *xattr_name;
 	int error;
 	/* xattr_resolve_name will do this for us if this is defined */
 #ifndef HAVE_XATTR_HANDLER_NAME
 	if (strcmp(name, "") == 0)
 		return (-EINVAL);
 #endif
+	if (ZFS_XA_NS_PREFIX_FORBIDDEN(name))
+		return (-EINVAL);
 	if (!(ITOZSB(ip)->z_flags & ZSB_XATTR))
 		return (-EOPNOTSUPP);
 
-	xattr_name = kmem_asprintf("%s%s", XATTR_USER_PREFIX, name);
-	error = zpl_xattr_get(ip, xattr_name, value, size);
-	kmem_strfree(xattr_name);
+	/*
+	 * Try to look up the name without the namespace prefix first for
+	 * compatibility with xattrs from other platforms.  If that fails,
+	 * try again with the namespace prefix.
```

Comment:
Since we're treating these both as equally valid formats, and I'd expect most pools not to move between platforms, why don't we check for the native format first to help minimize any potential performance impact.